apps/app/pages/404.tsx

```diff
@@ -1,9 +1,28 @@
-import { At, Container, ErrorStatus, Metadata } from "datagovmy-ui/components";
+import { Container, ErrorStatus, Metadata } from "datagovmy-ui/components";
 import { withi18n } from "datagovmy-ui/decorators";
 import { useTranslation } from "datagovmy-ui/hooks";
 import { Page } from "datagovmy-ui/types";
 import { GetStaticProps, InferGetStaticPropsType } from "next";
 import { useRouter } from "next/router";
+import { ComponentProps, useEffect, useState } from "react";
+
+const ActiveLink = ({ children, className, ...props }: ComponentProps<"a">) => {
+  const { asPath, isReady } = useRouter();
+  const [legacyPath, setLegacyPath] = useState<string>(props.href as string);
+
+  useEffect(() => {
+    // Check if the router fields are updated client-side
+    if (isReady) {
+      setLegacyPath(`https://archive.data.gov.my${asPath}`);
```

Comment:
does the old datagovmy also using the same route as we do for the dataset ? eg /data-catalogue/...

Reply:
don't think they are the same, i think it uses /dataset/...

Reply:
yeah what im saying is that if the redirection suppose to help point to a dataset in archieve datagovmy and its pointing to the wrong link anyway from the redirection, then wouldnt it be counter-intuitive ?